### PR TITLE
Geosys processing plugin

### LINKED ISSUE #24: EPIC 3.1: QGIS GEOSYS Processing Algorithm
**Expectation:** As a QGIS Processing toolbox user, I would like be able to incorporate the retrieval and analysis of GEOSYS products into my workflows.  

**Acceptance Criteria:**

- Given the GEOSYS plugin is installed and my credentials are set in the options dialog, when
I open the QGIS Processing toollbox, then the GEOSYS plugin will be available.
- Given that I am using the GEOSYS Processing Algorithm, when I define the request criteria
in the algorithm then the algorithm should successfully return the dataset requested.
- Given that a date is requested, when the user presses OK, then the most recent coverage
before the chosen date will be retrieved.
- Given that the GEOSYS algorithm may retrieve more than one coverage, when the user
presses OK, then the algorithm will merge the retrieved products as a single virtual raster and
add that raster to the QGIS map canvas.
- Given that a sensor is selected, when the user presses OK, then only coverages for that
sensor will be returned.
- Given that a polygon layer is selected, when the user presses OK, then the algorithm will
iteratively search GEOSYS for each polygon by passing the WKT string representation of the
polygon.

### PR DESCRIPTION
**Pull request by myarjunar**
fix #24 

Apr-07-2019 22-22-22

### CODE REVIEW

File: geosys/processing/geosys_processing_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from PyQt5.QtCore import QCoreApplication, QDate
+from PyQt5.QtWidgets import QDateEdit
+from processing.gui.wrappers import WidgetWrapper
+from qgis.core import (QgsProcessing,
```
Comment (timlinux) [inline geosys/processing/geosys_processing_algorithm.py]: pull left

File: geosys/processing/geosys_processing_algorithm.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+__copyright__ = "Copyright 2019, Kartoza"
+__license__ = "GPL version 3"
+__email__ = "rohmat@kartoza.com"
+__revision__ = "$Format:%H$"
+
+
+class DateWidgetWrapper(WidgetWrapper):
+    """WidgetWrapper for QgsProcessingParameterString that create and manage
+    a QDateEdit widget.
```
Comment (timlinux) [inline geosys/processing/geosys_processing_algorithm.py:62]: First sentance of docstring should fit in a single line

File: geosys/processing/geosys_processing_algorithm.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        date = QDate.fromString(value, 'yyyy-MM-dd')
+        self.date_edit.setDate(date)
+
+    def value(self):
+        """Override method."""
+        return self.date_edit.date().toString('yyyy-MM-dd')
+
+
+class MapCoverageDownloader(QgsProcessingAlgorithm):
+    """
```
Comment (timlinux) [inline geosys/processing/geosys_processing_algorithm.py]: First line of docstring should fit in a single line followed by two line breaks then extended description.

File: geosys/processing/geosys_processing_algorithm.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+    def name(self):
+        """
+        Returns the algorithm name, used for identifying the algorithm. This
+        string should be fixed for the algorithm, and must not be localised.
+        The name should be unique within each provider. Names should contain
+        lowercase alphanumeric characters only and no spaces or other
+        formatting characters.
+        """
+        return 'create_coverage_map'
```
Comment (timlinux) [inline geosys/processing/geosys_processing_algorithm.py]: maybe namespace this with geosys e.g. geosys_create_coverage_map ?

**Review by timlinux — APPROVED** [review]

**Comment by AndreLester** [comment]
LGTM

**Pull request merged**